vlm/src/test/scala/geotrellis/contrib/vlm/geotiff/GeoTiffRasterSourceMultiThreadingSpec.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm.geotiff
+
+import geotrellis.contrib.vlm.Resource
+
+import geotrellis.proj4.CRS
+import geotrellis.raster._
+import geotrellis.raster.resample._
+import geotrellis.util._
+
+import org.scalatest.AsyncFunSpec
+
+import scala.concurrent.{Future, ExecutionContext}
+import scala.concurrent.duration.Duration
+
+
+class GeoTiffRasterSourceMultiThreadingSpec extends AsyncFunSpec {
+  val url = Resource.path("img/aspect-tiled.tif")
+  val source: GeoTiffRasterSource = new GeoTiffRasterSource(url)
+
+  implicit val ec = ExecutionContext.global
+
+  describe("GeoTiffRasterSource should be threadsafe") {
+    it("read") {
+      val res: List[Future[Option[Raster[MultibandTile]]]] = (0 to 100).toList.map { _ => Future {
+        source.read()
+      } }
+
+      val fres: Future[List[Raster[MultibandTile]]] = Future.sequence(res).map(_.flatten)
+
+      fres.map { rasters => assert(rasters.size == 101) }
+    }
+
+    it("readBounds") {
+      val res: List[Future[Iterator[Raster[MultibandTile]]]] = (0 to 100).toList.map { _ => Future {
+        source.readBounds(List(source.gridBounds), 0 until source.bandCount toSeq)
```

Comment:
`source.read` should be there along with `readBounds` spec. Since it will cover a different call. 

The idea is to check all overloads and all unique calls.